### ISSUE TITLE
How can I override a uv install with an editable package that will stay up to date on edits

### ISSUE THREAD

**Issue by adiberk** [astral-sh/uv#10004]
I want to build my docker container as I normally do and then on run, install an editable version of a package so that I can make sure to get updates from that package

However, this doesn't seem to be working

In my dockerfile I have this code
```
WORKDIR /virtualenv
COPY pyproject.toml uv.lock ./
RUN --mount=type=ssh \
    if [ "$ENV" = "unittest" ]; then \
    echo 'Installing UT' && uv sync --extra ut --frozen --no-install-project; \
    elif [ "$ENV" = "dev" ]; then \
    echo 'Installing DEV' && uv sync --extra dev --frozen --no-install-project; \
    else \
    echo 'Installing REG' && uv sync --frozen --no-install-project; \
    fi
# create /app directory and chown to to node user or else it will be owned by root
WORKDIR /app
COPY --exclude=**/.* --exclude=*.lock --exclude=*.toml . /app

ENV VIRTUAL_ENV=/virtualenv/.venv
ENV PATH="/virtualenv/.venv/bin:$PATH"
ENV PYTHONHASHSEED=0
```
When I run my compose I try this
`uv pip install -e /py-authorization /our-common --config-settings editable_mode=strict  && python -m debugpy --listen 0.0.0.0:10001 -m gunicorn -c python:config --bind=0.0.0.0:7680 place:app --log-level=debug --reload`

While this does make sure my running container gets the latest local code from "our-common" it is not updating after I edit that code

Before when we used pip this works just fine

**Comment by charliermarsh** [comment]
Have you confirmed that everything is using the same virtual environment? How did you confirm that the code is "not updating"?

**Comment by adiberk** [comment]
> Have you confirmed that everything is using the same virtual environment? How did you confirm that the code is "not updating"?

@charliermarsh 
I added a print statement in the local package while my service was running. The statement still wasn't getting hit. I have to restart (and rerun that uv install command)

Also, if I debug and enter the local package, I can see it isn't showing latest data

**Comment by adiberk** [comment]
@charliermarsh I actually tried this with a differernt project that install directly to system python (in docker) and it worked just fine.

So I think I know the issue
`which python` shows the path to virtual env
BUT
`which pip` shows path to `/usr/local/bin/pip` which seems like an issue

**Comment by zanieb** [comment]
It sounds like you want to add pip as a dependency of your project? Or do `uv pip install pip` so it's available in the environment?

Your example uses `uv pip` not `pip` though.

**Comment by adiberk** [comment]
Right so I am happy to use uv pip and not just "pip"
But it seems it was working with uv pip the way it is working with just regular pip

**Comment by zanieb** [comment]
If you install it with regular pip it seems like it'd be outside the virtual environment though?

**Comment by adiberk** [comment]
Agreed which is why I only did that when I installed the packages using the system python install docker and **not** a virtual env. This helped me verify that it might be some sort of issue with the virtual env and using `uv pip install -e` in the run compose command

**Comment by zanieb** [comment]
I see. Sorry I didn't fully understand your previous comment.

I think we'll need a complete minimal example to help here.
